src/Futhark/CLI/REPL.hs

```diff
@@ -405,6 +417,41 @@ typeCommand = genTypeCommand parseExp T.checkExp $ \(ps, e) ->
 mtypeCommand :: Command
 mtypeCommand = genTypeCommand parseModExp T.checkModExp $ pretty . fst
 
+parseForPrint :: T.Text -> Either T.Text ([T.Text], [T.Text])
+parseForPrint printStr
+  | not balanced =
+      Left
+        "Parse error on print string. Likely due to mismatched braces {}."
+  | otherwise = Right (strs, exps)
+  where
+    firstSplit = T.split (== '{') printStr
+    splits = map (T.split (== '}')) $ tail firstSplit
+    balanced =
+      (T.count "{" printStr == T.count "}" printStr)
+        && all (\x -> length x == 2) splits
+    strs = head firstSplit : map last splits
+    exps = map head splits
+
+printCommand :: Command
+printCommand input = do
+  case parseForPrint input of
+    Left err -> liftIO $ T.putStrLn err
+    Right (strs, exps) -> do
+      prompt <- getPrompt
+      case partitionEithers $ map (parseExp prompt) exps of
+        ((SyntaxError _ err) : _, _) -> liftIO $ T.putStr err
+        (_, uncheckedExps) -> do
+          vals <- mapM onExp uncheckedExps
```

Comment:
I have a hunch that continuing to evaluate later expressions, after an early one has failed, would be confusing.  I suggest stopping at the first error.

Reply:
That sounds like a good idea, I added another commit resolving that, and also providing a *slightly* better error message if there is an error in an expression.